site/src/components/service-message-list.tsx

```diff
@@ -0,0 +1,187 @@
+            </tr>
+          </thead>
+          <tbody>
+            {Object.entries(data.properties as Record<string, JsonSchema>).map(
+              ([propertyName, propertySchema]) => (
+                <tr key={propertyName}>
+                  <td>
+                    <Typography variant="bpSmallCopy">
+                      <MdxCode>{propertyName}</MdxCode>
+                    </Typography>
+                  </td>
+                  <td>
+                    <Typography variant="bpMicroCopy">
+                      <DataType propertySchema={propertySchema} />
+                    </Typography>
+                  </td>
+                  <td>
+                    {((data.required as string[]) ?? []).includes(
+                      propertyName,
+                    ) ? (
+                      <Typography
+                        variant="bpMicroCopy"
+                        sx={{ fontWeight: 700 }}
+                      >
+                        yes
+                      </Typography>
+                    ) : (
+                      <Typography variant="bpMicroCopy">no</Typography>
+                    )}
+                  </td>
+                  <td>
+                    <Typography variant="bpMicroCopy">
+                      {propertySchema.description}
+                    </Typography>
+                  </td>
+                </tr>
+              ),
+            )}
+          </tbody>
+        </MdxTable>
+      </>
+    );
+  }
+  return (
+    <MdxCode>
+      <DataType propertySchema={data} />
+    </MdxCode>
+  );
+};
+
+const ServiceMessage: React.VoidFunctionComponent<{
+  message: ServiceMessageDefinition;
+}> = ({ message }) => {
+  const MdxCode = mdxComponents.code!;
+  const MdxLi = mdxComponents.li!;
+  const MdxUl = mdxComponents.ul!;
+  const MdxH5 = mdxComponents.h5!;
+  const MdxA = mdxComponents.a!;
+
+  const {
+    data,
+    description,
+    errorCodes,
+    messageName,
+    respondedToBy,
+    sentOnInitialization,
+    source,
+  } = message;
+
+  return (
+    <Box sx={{ mb: 4 }}>
+      <MdxH5>
+        <MdxCode>{messageName}</MdxCode>
+      </MdxH5>
+      <Typography sx={{ mb: 1, mt: 1 }} variant="bpSmallCopy" component="div">
+        {description}
+      </Typography>
+      <MdxUl>
+        <MdxLi>
+          <MdxCode>source</MdxCode>:{" "}
+          <Typography variant="bpSmallCopy" sx={{ fontWeight: 600 }}>
+            {source}
+          </Typography>
+        </MdxLi>
+        <MdxLi>
+          <MdxCode>data</MdxCode>: <ServiceMessageData data={data} />
+        </MdxLi>
+        <MdxLi>
+          <MdxCode>errorCodes</MdxCode>:{" "}
+          <Typography variant="bpSmallCopy" sx={{ fontWeight: 600 }}>
+            {errorCodes && errorCodes.length > 0
+              ? errorCodes.map((code, index) => (
+                  <>
+                    {index > 0 ? ", " : ""}
+                    <MdxCode key={code}>{code}</MdxCode>
+                  </>
+                ))
```

Comment:
I could be wrong, but doesn't the key have to be on the fragment here?
```suggestion
              ? errorCodes.map((code, index) => (
                  <React.Fragment key={code}>
                    {index > 0 ? ", " : ""}
                    <MdxCode>{code}</MdxCode>
                  </React.Fragment>
                ))
```